codes/tenant/controllers/tenant_controller.go

```diff
@@ -25,19 +25,27 @@ type TenantReconciler struct {
 	Log      logr.Logger
 	Scheme   *runtime.Scheme
 	Recorder record.EventRecorder
-	stopCh   <-chan struct{}
 }
 
 //! [rbac]
 // +kubebuilder:rbac:groups=multitenancy.example.com,resources=tenants,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=multitenancy.example.com,resources=tenants/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=multitenancy.example.com,resources=tenants/finalizers,verbs=update
```

Comment:
I cannot find `finalizers` subresource support for CRDs.
https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#subresources

Reply:
It seems to be used in the `OwnerReferencesPermissionEnforcement` admission controller.
(However, it's not enabled by default.)

https://github.com/kubernetes-sigs/kubebuilder/pull/1688
https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement
https://github.com/kubernetes/kubernetes/blob/87984d84d18c7816835e5efedebb17ed13aa2509/plugin/pkg/admission/gc/gc_admission.go

Reply:
Understood.